### PR TITLE
MBS-11461 hide finalizer in test-runner:client

### PR DESCRIPTION
**Pull request by dsvoronin**
remove InstrumentationTestsAction
rename InstrumentationTestsActionParams -> RunnerInputParams

### CODE REVIEW

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt [hunk 1]
```diff
@@ -1,47 +1,229 @@
...
+            metricsConfig = metricsConfig,
+            reportFactory = reportFactory,
+            timeProvider = timeProvider
+        )
+
+        val androidDebugBridgeProvider = createAndroidDebugBridgeProvider(params.loggerFactory)
+
+        val emulatorsLogsReporterProvider = createEmulatorsLogsReporterProvider(params)
+
+        val testRunnerFactoryProvider = createTestRunnerFactoryProvider(
```
Comment (sergioboi) [inline subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt]: I think that is not what `TestShecdulerFactory` should do
`TestSchedulerFactory` must only create the `TestScheduler` and get `testRunnerFactoryProvider` as an input parameter

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt [hunk 2]
```diff
@@ -1,47 +1,229 @@
...
+            finalizer = finalizerFactory.create(),
         )
     }
 
-    private fun createTestSuiteProvider(): TestSuiteProvider = TestSuiteProvider.Impl(
-        report = report,
-        targets = params.instrumentationConfiguration.targets,
-        reportSkippedTests = params.instrumentationConfiguration.reportSkippedTests,
-        filterFactory = FilterFactory.create(
+    private fun createFinalizerFactory(
```
Comment (sergioboi) [inline subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt]: `TestShedulerFactory` must not know how to create a Finalizer
Make FinalizerFactory as an input parameter

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt [hunk 3]
```diff
@@ -1,47 +1,229 @@
...
+    }
+
+    private fun createStatsdSender(params: RunnerInputParams): StatsDSender {
+        return StatsDSender.Impl(
+            config = params.statsDConfig,
+            loggerFactory = params.loggerFactory
+        )
+    }
+
+    private fun createTestSuiteLoader(): TestSuiteLoader {
```
Comment (sergioboi) [inline subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt]: Useless

**Review by sergioboi — CHANGES_REQUESTED** [review]

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt (hunk 2 above)
Comment (dsvoronin) [reply subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt]: Finalizer is now hidden in TestScheduler implementation, so i can only extract it like this:

```
TestSchedulerFactoryImpl(finalizerFactory (params) -> Filnalizer) {

   override fun create(params: RunnerInputParams): TestScheduler {
        val finalizerFactory = finalizerFactory.invoke(params)
   }
}
```

which is odd

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt (hunk 3 above)
Comment (dsvoronin) [reply subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt]: hides implementation, i had plans to look why Implementation class is here, and not hidden in some module

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt (hunk 1 above)
Comment (dsvoronin) [reply subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt]: so we need something like TestSchedulerFactoryFactory, or entrypoint for runner?

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt (hunk 3 above)
Comment (sergioboi) [reply subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt]: I don't want to debate lets discuss it at the call

**Review by sergioboi — APPROVED** [review]

**Pull request merged**